File: Tema2/Ejercicio06/ejercicio6.py
import math
from random import randint


class ejercicio6:

    def ecuacion():
        a = input("Vamos a realizar una ecuacion, introduce tres numeros para realizarla: a = ")
        b = input("Otro numero más: b = ")
        c = input("Un ultimo número: c = ")
        discriminante = pow(b,2) - (4 * a * c)
        if (a == 0 or b == 0 or c == 0):
            print("La solución es 0")
        elif (discriminante == 0):
            x = -b / (2 * a)
            print("La solución es: " + x)
        elif (discriminante > 0):
            x = (-b - (math.sqr(discriminante)) /(2 * a))
            x1 = (-b + (math.sqr(discriminante)) /(2 * a))
            print("Las soluciones son: " + x + " y " + x1)
        else:
            print ("No hay solucion REAL")


    if __name__ == '__main__':
        ecuacion()
        

File: Tema2/Ejercicio06/ejercicio6.java
import java.util.Scanner;

public class ejercicio6 {
    public static void main(String[] args) {

        Scanner scanner = new Scanner(System.in);
        System.out.println("Vamos a solucionar una ecuación de segundo grado: ax^2 + bx + c = 0");
        System.out.println("Introduce un numero para A: ");
        double a = scanner.nextDouble();
        System.out.println("Introduce un numero para B: ");
        double b = scanner.nextDouble();
        System.out.println("Introduce un numero para C: ");
        double c = scanner.nextDouble();
        double discriminante = Math.pow(b,2) - (4 * a * c);
        boolean funcionando = true;

        

            if (a == 0 || b == 0 || c == 0){
                System.out.println("La solución es: 0");
                
            }else if ((discriminante) == 0){
                double x = -b / (2 * a);
                System.out.println("La solución es:" + x);
            }else if ((discriminante) > 0){
                double x = (-b - (Math.sqrt(discriminante))) / (2*a);
                double x1 = (-b + (Math.sqrt(discriminante))) / (2*a);
                System.out.println("Las soluciones son:" + x +" y "+ x1);
            }else{
                System.out.println("No hay solucion REAL");
            
        }  
    }
}
